### PR TITLE
Add a check for empty params for `url join`

### PR DESCRIPTION
**Pull request by gdhuper**
# Description

Fix for #9347 

# User-Facing Changes

# Tests + Formatting

- [x] Add unit tests
- [x] Run all cargo tests + fmt commands

# After Submitting

### CODE REVIEW

**Comment by fdncred** [comment]
seems like a reasonable fix.

**Comment by gdhuper** [comment]
> seems like a reasonable fix.

File: crates/nu-command/tests/commands/url/join.rs [hunk 1]
```diff
@@ -366,3 +366,21 @@ fn url_join_with_fragment_and_params() {
...
+#[test]
+fn url_join_with_empty_params() {
+    let actual = nu!(
+        cwd: ".", pipeline(
+            r#"
+            {
+                "scheme": "https",
+                "host": "localhost",
+                "path": "/foo",
+            } | url join
```
Comment (amtoine) [inline crates/nu-command/tests/commands/url/join.rs:379]: actually this was already equal to `https://localhost/foo` on the current `main` branch, so i think this is not the thing to test :thinking: 

however,
```bash
{
    scheme: https,
    host: example.com,
    path: /foo,
    params: {}
} | url join
```
is equal, from `cargo run`, to
- `https://example.com/foo?` on 82e6873702f2e3588e887915e2c5acdb54194bc0, i.e. latest `main` right now
- `https://example.com/foo` on the tip of this PR branch

so it looks to me this should be the test :yum: 

> **Note**
> the record was generated from `https://example.com/foo | url parse` :+1:

**Review by amtoine — CHANGES_REQUESTED** [review]
coool :ok_hand: 

looks like the bug is gone however, as said in a review thread, i think the test is not enough to capture the fix :thinking:

File: crates/nu-command/tests/commands/url/join.rs (hunk 1 above)
Comment (gdhuper) [reply crates/nu-command/tests/commands/url/join.rs]: Thanks for catching this. Fixed the unit test.

File: crates/nu-command/tests/commands/url/join.rs [hunk 2]
```diff
@@ -366,3 +366,22 @@ fn url_join_with_fragment_and_params() {
...
+                "host": "localhost",
+                "path": "/foo",
+                "params": {}
+            } | url join
+            "#
+        )
+    );
+
+    assert_eq!(actual.out, "https://localhost/foo");
+    assert_ne!(actual.out, "https://localhost/foo?")
```
Comment (amtoine) [inline crates/nu-command/tests/commands/url/join.rs:386]: here, if we get to the `assert_ne` line, this means that `actual.out` **is** equal to `"https://localhost/foo"` and thus it's **impossible** for this variable to equal `"https://localhost/foo?"` :smirk: 

i think this second `assert` is redundant :relieved:

Comment (gdhuper) [reply crates/nu-command/tests/commands/url/join.rs]: Removed.

**Review by amtoine — APPROVED** [review]
yeaaaah, i think we can land this :star_struck: 
thanks @gdhuper :blush:

**Pull request merged**